### PR TITLE
Add Java API 'getRef' to 'Terminated' (#22985)

### PR DESCRIPTION
**Pull request by raboof**
Refs #22985

### CODE REVIEW

File: akka-typed/src/main/scala/akka/typed/MessageAndSignals.scala [hunk 1]
```diff
@@ -63,4 +63,8 @@ final case class Terminated(ref: ActorRef[Nothing])(failed: Throwable) extends S
   def wasFailed: Boolean = failed ne null
   def failure: Throwable = failed
   def failureOption: Option[Throwable] = Option(failed)
+
+  /** Java API */
+  def getRef(): ActorRef[Nothing] = ref
```
Comment (patriknw) [inline akka-typed/src/main/scala/akka/typed/MessageAndSignals.scala]: at some places we use `ActorRef[Void]` in Java API, should that be the case here also?

**Review by patriknw — COMMENTED** [review]
let's merge this after the release, it's not critical `ref()` can still be used from Java

File: akka-typed/src/main/scala/akka/typed/MessageAndSignals.scala (hunk 1 above)
Comment (ktoso) [inline akka-typed/src/main/scala/akka/typed/MessageAndSignals.scala]: +1, Java must not see Nothing :)

Comment (raboof) [inline akka-typed/src/main/scala/akka/typed/MessageAndSignals.scala]: Hmm yeah `Nothing` is a scala-ism ofc... though wouldn't `ActorRef[Void]` suggest you can send it a `null`? Probably fine though, we have lots of API's that strictly speaking accept `null` but will definitely error ;).

**Review by johanandren — APPROVED** [review]

**Pull request merged**